### PR TITLE
Build an SVG element under the name the parser gives it

### LINKED ISSUE #1106: An SVG element built on the client loses its tag name's case
HTML source carries no tag-name case. The tokenizer lowercases every tag name before anything else sees it, and inside `<svg>` the tree builder then restores the spelling the SVG spec uses. So these are all the same markup:

```
<linearGradient>   ->  linearGradient
<lineargradient>   ->  linearGradient
<LINEARGRADIENT>   ->  linearGradient
<div> <DIV> <DiV>  ->  div
```

The client creates elements under the name the template wrote, and `createElementNS` restores nothing:

```js
document.createElementNS("http://www.w3.org/2000/svg", "linearGradient"); // a gradient
document.createElementNS("http://www.w3.org/2000/svg", "lineargradient"); // not a gradient
```

So this template gives one element on a page the server rendered and a different one on a page the client built:

```html
<svg>
  <defs><lineargradient id="g"><stop offset="0" /></lineargradient></defs>
  <rect fill="url(#g)" width="10" height="10" />
</svg>
```

```
server-rendered             client-built
svg                         svg
  defs                        defs
    linearGradient              lineargradient
      stop                        stop
  rect                        rect
```

Both are in the SVG namespace, but `lineargradient` is not an SVG element, so the rect's `url(#g)` has nothing to resolve to. The same goes for `clipPath`, `textPath`, `foreignObject` and the `fe*` filter primitives - every SVG element whose name carries a capital.

The client builds rather than adopts whenever the markup was not already on the page: after a navigation, or when a block brings the element in. So the page can load correctly and then lose the gradient when something toggles.

Related to #1024, which is the same case difference on the adoption side.

### PR DESCRIPTION
**Pull request by bartblast**
Closes #1106

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved dynamic HTML tag rendering by normalizing tag names consistently.
  - Corrected SVG tag casing, including `linearGradient`, to ensure elements render and update properly.
  - Improved handling of uppercase HTML tags and dynamically created SVG elements.

- **Tests**
  - Added coverage for dynamic HTML and SVG tags, self-closing elements, and edge-case tag names.

### CODE REVIEW

File: lib/hologram/template/helpers.ex [hunk 1]
```diff
@@ -19,6 +19,68 @@ defmodule Hologram.Template.Helpers do
...
+    "fespecularlighting" => "feSpecularLighting",
+    "fespotlight" => "feSpotLight",
+    "fetile" => "feTile",
+    "feturbulence" => "feTurbulence",
+    "foreignobject" => "foreignObject",
+    "glyphref" => "glyphRef",
+    "lineargradient" => "linearGradient",
+    "radialgradient" => "radialGradient",
+    "textpath" => "textPath"
+  }
```
[the remark this answers is not in the document]
Comment (bartblast) [reply lib/hologram/template/helpers.ex:64]: Rejected - the omission is deliberate.

`feDropShadow` is missing from the browser's own list. That list lives in the HTML parsing spec's [adjust SVG tag name](https://html.spec.whatwg.org/multipage/parsing.html#adjust-svg-tag-name) step and was written for SVG 1.1, before the element existed, so no HTML markup can produce a `feDropShadow` element at all - `<fedropshadow>` and `<feDropShadow>` both parse to `fedropshadow`. Verified against jsdom's parser: 36 names are adjusted, and that one is not.

Adding the entry would make Hologram's client build an element the browser's parser never produces, so a client-built page would disagree with the server-rendered one for this single name. The point of the change is that the two halves agree, so the table copies the browser's list exactly.

The `feDropShadow` assertion in `helpers_test.exs` is not obsolete - it pins that decision, with a comment saying why, so the name reads as a choice rather than an oversight.
